### PR TITLE
fix(cloudwatch): `period` of each metric in `usingMetrics` for `MathExpression` is ignored

### PR DESCRIPTION
**Pull request by go-to-k**
### Issue # (if applicable)

Closes #<issue number here>.

### Reason for this change

The `usingMetrics` property (`Record<string, IMetric>`) in `MathExpressionProps` has Metric objects with a `period`.

On the other hand, in the `MathExpression` construct, the `period` of each metric in the `usingMetrics` is ignored and instead overridden by the `period` of the `MathExpression`. Even if the `period` of the `MathExpression` is not specified, it is overridden by its default value.

https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-cloudwatch/lib/metric.ts#L606-L608

However the statement is not written in the JSDoc.

https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-cloudwatch/lib/metric.ts#L566

```ts
new MathExpression({
  expression: "m1+m2",
  label: "AlbErrors",
  usingMetrics: {
    m1: metrics.custom("HTTPCode_ELB_500_Count", {
      period: Duration.minutes(1), // <- ignored and overridden by default value `Duration.minutes(5)` of `period` in the `MathExpressionOptions`
      statistic: "Sum",
      label: "HTTPCode_ELB_500_Count",
    }),
    m2: metrics.custom("HTTPCode_ELB_502_Count", {
      period: Duration.minutes(1), // <- ignored and overridden by default value `Duration.minutes(5)` of `period` in the `MathExpressionOptions`
      statistic: "Sum",
      label: "HTTPCode_ELB_502_Count",
    }),
  },
}),
```

### Description of changes

The current documentation could be confusing to users, so add this description. Also added warnings in the situation.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by kaizencc — DISMISSED** [review]
In addition to this, what if we added a warning annotation anytime we find ourselves overwriting a different value? it's a little sinister to override a value and not tell the user when we do it.

File: packages/aws-cdk-lib/aws-cloudwatch/lib/metric.ts
Comment (go-to-k) [reply packages/aws-cdk-lib/aws-cloudwatch/lib/metric.ts]: I've rewritten it, what do you think?

https://github.com/aws/aws-cdk/pull/30986/commits/13f3459476dd59dd0558800e3c26b58aaaa7d125

Comment (go-to-k) [reply packages/aws-cdk-lib/aws-cloudwatch/lib/metric.ts]: I've rewritten it, what do you think?

https://github.com/aws/aws-cdk/pull/30986/commits/b8b2ae083413f0a3a4bd7b0675e32c92ee1ccef8

**Comment by go-to-k** [comment]
@kaizencc 

Thanks for your review.

> In addition to this, what if we added a warning annotation anytime we find ourselves overwriting a different value? it's a little sinister to override a value and not tell the user when we do it.

I added: https://github.com/aws/aws-cdk/pull/30986/commits/712e937e0a5c5fd69e6e582320cdff5cd839ea81

I have two questions:

1. Should I change the PR title from `docs(cloudwatch): ...` to something else? (I thought this PR was not docs because warning process added.) Or, should PRs be separated?

2. We can't warn only if the period in `usingMetrics` is set to the same value as the default value (`Duration.minutes(5)`), is there a problem?

It is correct usage to not specify a value for period (in usingMetrics), so we do not want to raise a warning in that case. However, if the period is not specified, it is overwritten with a default value, so we cannot determine whether the user has specified that value or if it has been overwritten. Therefore, warnings cannot be raised only in this case.

https://github.com/aws/aws-cdk/pull/30986/commits/712e937e0a5c5fd69e6e582320cdff5cd839ea81#diff-289bad9fd897ab7808c177ad955a23faf18615ff8434f82b50fa0aeb54985de6R941

for example: 

```ts
new MathExpression({
  expression:  'm1',
  label: 'AlbErrors',
  usingMetrics: {
    m1: metrics.custom('HTTPCode_ELB_500_Count', {
      period: Duration.minutes(5), // <- This is equal to the default value for `period`. So it is the same as not having specified.
      statistic: 'Sum',
      label: 'HTTPCode_ELB_500_Count',
    }),
  },
  period: Duration.minutes(3),
});
```

File: packages/aws-cdk-lib/aws-cloudwatch/lib/metric.ts
Comment (kaizencc) [reply packages/aws-cdk-lib/aws-cloudwatch/lib/metric.ts]: the example is awesome! thanks so much

**Comment by go-to-k** [comment]
Hi, @kaizencc 

Have you had a chance to look at the comment?

https://github.com/aws/aws-cdk/pull/30986#issuecomment-2331341259

**Comment by go-to-k** [comment]
@kaizencc

Could you please take a look at the comment? Do you have any good ideas to share?

**Comment by go-to-k** [comment]
@kaizencc Could you take a look at it again?

File: packages/aws-cdk-lib/rosetta/aws_cloudwatch/default.ts-fixture
Comment (kaizencc) [inline packages/aws-cdk-lib/rosetta/aws_cloudwatch/default.ts-fixture:5]: i think this is unnecessary?

**Review by kaizencc — COMMENTED** [review]
hi @go-to-k  :). sorry for the super long delay. to answer your questions:

1) i think the title should be now a fix. i'll leave it to you to update.
2) i think this is ok. it's better than nothing.

happy to approve once you update! i think the change to `default.ts-fixture` is unnecessary but i might be wrong.

File: packages/aws-cdk-lib/aws-cloudwatch/lib/metric.ts
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-cloudwatch/lib/metric.ts:237]: here

File: packages/aws-cdk-lib/rosetta/aws_cloudwatch/default.ts-fixture
Comment (go-to-k) [reply packages/aws-cdk-lib/rosetta/aws_cloudwatch/default.ts-fixture:5]: Thank you @kaizencc !

I think it is necessary here, could you check it?

https://github.com/aws/aws-cdk/pull/30986/files#r1868706284

Comment (go-to-k) [reply packages/aws-cdk-lib/rosetta/aws_cloudwatch/default.ts-fixture:5]: https://github.com/aws/aws-cdk/pull/30986#discussion_r1868706284

**Comment by go-to-k** [comment]
Exemption Request: This PR could be covered by unit tests.

**Comment by go-to-k** [comment]
@kaizencc 

I also changed the PR title to fix.

File: packages/aws-cdk-lib/aws-cloudwatch/lib/metric.ts
Comment (kaizencc) [reply packages/aws-cdk-lib/aws-cloudwatch/lib/metric.ts:237]: got it! thanks :)

**Review by kaizencc — APPROVED** [review]

**Comment by kaizencc** [comment]
@mergifyio refresh

**Comment by go-to-k** [comment]
@Mergifyio update

**Pull request merged**
